app/models/piece.rb

```diff
@@ -56,5 +56,126 @@ def count_moves
     update(has_moved: true)
   end
 
+  def is_vertically_obstructed?(new_x, new_y)
+    x1 = self.x_position
+    y1 = self.y_position
+    if (y1 < new_y)
+      small_y = y1
+      big_y = new_y
+    else
+      small_y = new_y
+      big_y = y1
+    end
+    i = small_y + 1
+
+    while i < big_y  do
+      if ($board[x1][i] === nil)
```

Comment:
@Nikkovichert all the `$board[a][b]` lines are basically trying to find a piece at the position, correct? I'd suggest adding a method like `piece_present_at?(x,y)` and this method will call `game.pieces.find(...)`.. we don't really have `$board`..